website/blog/2021/06/28/why-we-need-Apache-APISIX.md

```diff
@@ -10,118 +10,117 @@ keywords:
 
-Finally, the bottom right part of the service grid, for the service grid has formed a choice istio (CP) + Envoy (DP). Later we see the Alibaba open source mosn, in a nutshell: Golang version of envoy.
+Finally, the service mesh has formed a choice istio (CP) and Envoy (DP) in the bottom right part of the service mesh. Later we see the Alibaba open-source mosn, in a nutshell: the Golang version of envoy.
 
 ![APISIX architecture](https://static.apiseven.com/202108/1646300450364-7318620a-0876-472e-8d7b-880c0f4a7ac2.jpg)
 
-Reviewing the previous architecture evolution diagram, I believe many students have found out where the problem is. From top to bottom, from left to right, for different scenarios, we finally "reasonably" introduced various components to solve our problems, the architect's rule of survival: choose the most suitable at the moment.
+Reviewing the previous architecture evolution diagram, I believe you have found out where the problem is. From top to bottom, from left to right, for different scenarios, we finally "reasonably" introduced various components to solve our problems, the architect's rule of survival: choose the most suitable at the moment.
 
-When we have few tools at hand, we always have to compromise between functionality, dynamics, performance, etc. We have long been accustomed to and even numb to the rapid development of IT technology, are they still the most appropriate solution today?
+When we have few tools, we always compromise between functionality, dynamics, performance, etc. Although we have long been accustomed to and even numb to the rapid development of IT technology, are they still the most appropriate solution today?
 
 ![Nginx](https://static.apiseven.com/202108/1646300822188-c239210d-ee4c-4025-a2a5-9e40da1e7863.jpg)
 
-As you can see, these are NGINX drawbacks, such as NGINX's low activity community. While we could invest more resources at the corporate level, his community is really unfriendly, and how unfriendly is it? As you can see in the picture above, the NGINX repository in Github is only a mirror, the issue function is closed, it is impossible to submit an issue, and even if you submit a PR the official will not merge it.
+As you can see, these are NGINX drawbacks, such as NGINX's low activity community:
+The NGINX repository in Github is only a mirror.
+The GitHub Issue feature is disabled.
+It is impossible to submit an issue.
+The official team will not merge even if you submit a PR.
+In addition, NGINX is weak in its routing. For example, if I want to do a canary release based on a request parameter such as id, you will find that NGINX is entirely unable to achieve it. So we can see a lot of small open-source projects. As long as the above canary release scenario is solved, it can be an independent open source project. In addition, gRPC calls are becoming increasingly popular in microservice calls, but NGINX support for it is only "simple to use."
```

Comment:
```suggestion
As you can see, these are NGINX drawbacks, such as NGINX's low activity community:

- The NGINX repository in Github is only a mirror.
- The GitHub Issue feature is disabled.
- It is impossible to submit an issue.
- The official team will not merge even if you submit a PR.

In addition, NGINX is weak in its routing. For example, if I want to do a canary release based on a request parameter such as id, you will find that NGINX is entirely unable to achieve it. So we can see a lot of small open-source projects. As long as the above canary release scenario is solved, it can be an independent open source project. In addition, gRPC calls are becoming increasingly popular in microservice calls, but NGINX support for it is only "simple to use."
```